FaceDetectionServer/client/multi_client.py

```diff
@@ -82,18 +83,64 @@ def __init__(self, host, port):
         self.filename_list_index = 0
         self.json_params = None
         self.base64 = False
+        self.video = None
+        self.resize = True
+        self.resize_long = 240
+        self.resize_short = 180
+        self.skip_frames = 1
         logger.debug("host:port = %s:%d" %(self.host, self.port))
 
     def start(self):
-        logger.debug("image file(s) %s %s" %(self.image_file_name, self.filename_list))
-
-    def next_file_name(self):
-        """ If the filename_list array is has more than 1, get the next value. """
-        if len(self.filename_list) > 1:
-            self.filename_list_index += 1
-            if self.filename_list_index >= len(self.filename_list):
+        logger.debug("image file(s) %s" %(self.filename_list))
+        video_extensions = ('mp4','avi', 'mov')
+        if self.filename_list[0].endswith(video_extensions):
+            logger.debug("It's a video")
+            self.image_file_name = self.filename_list[0]
+            self.video = cv2.VideoCapture(self.image_file_name)
+
+    def get_next_image(self):
+        if self.video is not None:
+            for x in range(self.skip_frames):
+                ret, image = self.video.read()
```

Comment:
What does the read() function do? Does that append a certain number of video frames to image? I'm also a little confused what self.video is.

Reply:
self.video is an OpenCV VideoCapture instance. It could be fed by a webcam, or in this case, a video file. read() gets the next single frame of the video. When "ret" is non-True, we have reached the end of the video, or encountered an error.